Packages/com.unity.inputsystem/InputSystem/Plugins/UI/InputSystemUIInputModuleEditor.cs

```diff
@@ -122,17 +130,34 @@ public override void OnInspectorGUI()
             }
 
             var numActions = s_ActionNames.Length;
-            for (var i = 0; i < numActions; i++)
-            {
-                if (m_AvailableActionsInAsset == null)
-                    continue;
 
-                var index = Array.IndexOf(m_AvailableActionsInAsset, m_ReferenceProperties[i].objectReferenceValue) + 1;
-                EditorGUI.BeginChangeCheck();
-                index = EditorGUILayout.Popup(s_ActionNiceNames[i], index, m_AvailableActionsInAssetNames);
+            if ((m_AvailableActionReferencesInAssetDatabase != null && m_AvailableActionReferencesInAssetDatabase.Length > 0) || m_ActionsAsset.objectReferenceValue == null)
+            {
+                for (var i = 0; i < numActions; i++)
+                {
+                    var index = Array.IndexOf(m_AvailableActionReferencesInAssetDatabase,
+                        m_ReferenceProperties[i].objectReferenceValue) + 1;
+                    EditorGUI.BeginChangeCheck();
+                    index = EditorGUILayout.Popup(s_ActionNiceNames[i], index, m_AvailableActionsInAssetNames);
+
+                    if (EditorGUI.EndChangeCheck())
+                        m_ReferenceProperties[i].objectReferenceValue =
+                            index > 0 ? m_AvailableActionReferencesInAssetDatabase[index - 1] : null;
+                }
+            }
+            else
+            {
+                // Somehow we have an asset but no asset references from the database, pull out references manually and show them in read only UI
+                EditorGUILayout.HelpBox("Showing fields as read-only because current action asset seems to be created by a script and assigned programmatically.", MessageType.Info);
+                for (var i = 0; i < numActions; i++)
```

Comment:
Probably better in disabled block (`EditorGUI.BeginDisabled`/`EndDisabled` or `DisabledScope`) to actually make read-only. Bit confusing that the UI is actually not read-only. Also, this way the codepaths can probably remain the same.

Reply:
Made `EditorGUI.BeginDisabledGroup`, looks much better now 👍 

Haven't managed to combine both codepathes though, they do seem to be very different indeed, I'll keep them separate instead.